src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,665 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""
+==================================================================================
+  Iterative Allocation Process(IAP) Reference Model [R2-SGN-16].
+  Based on WINNF-TS-0112, Version V1.4.1, 16 January 2018.
+
+  This model provides functions to calculate post IAP allowed interference margin for 
+  the protection constraint of FSS, ESC, GWPZ and PPA. 
+
+  The main routines are:
+
+    performIap 
+    performIapForEsc
+    performIapForGwpz
+    performIapForPpa
+    performIapForFss
+    calculateApIapRef
+
+  The routines return a dictionary containing key as protection constraints 
+  (lat, long, low_freq, high_freq, entity_type) and value as the post IAP aggregate 
+  interference value in mW/RBW for each of the constraints.
+==================================================================================
+"""
+
+from reference_models.geo import utils
+from reference_models.interference import interference as interf
+import numpy as np
+import logging
+from collections import namedtuple
+from multiprocessing import Pool
+
+# Define Post IAP Grant information, i.e., a tuple with named fields of
+# 'eirp', 'interference'
+PostIAPGrantInfo = namedtuple('PostIAPGrantInfo', ['eirp', 'interference'])
+
+# Global Post IAP managed grants container
```

Comment:
This   post_iap_managed_grants dictionary is used for storing internal data during   IAP procedure execution. It is used just for    'A_SAS_P' variable in the equation. No other operations are performed   on the stored data. We thought it would be simpler and easier to use if it is   kept as a dictionary.

Reply:
I guess Kate refers to the fact that only rarely are global variable suitable. It usually bites you in the long term. 
When temp variable are needed for temporary data, this means that the overall run function would be better as a method in a class, which uses the instance variable to store its state.
Using global will create potential issues for exmaple for running in parallel things becomes impossible or hard.
Or cleaning up the state of the globals after a run so the globals are clean for the next run.